lib/protoparser/prometheus/parser.go

```diff
@@ -185,25 +198,86 @@ func (r *Row) unmarshal(s string, tagsPool []Tag, noEscapes bool) ([]Tag, error)
 	return tagsPool, nil
 }
 
-var rowsReadScrape = metrics.NewCounter(`vm_protoparser_rows_read_total{type="promscrape"}`)
+var protobufRowsReadScrape = metrics.NewCounter(`vm_protoparser_rows_read_total{type="promscrape",format="PrometheusProto"}`)
+
+func unmarshalProtobuf(dst []Row, s []byte, tagsPool []Tag, errLogger func(s string)) ([]Row, []Tag) {
+	dstLen := len(dst)
+	var r ProtoRequest
+	if err := r.UnmarshalProtobuf(s); err != nil {
+		return dst, tagsPool
+	}
+	for _, mf := range r.Families {
+		row := Row{
+			Metric: mf.Name,
+		}
+		for _, m := range mf.Metrics {
+			row.Timestamp = m.Timestamp
+			tagsCount := len(row.Tags)
+			row.Tags = make([]Tag, len(m.Labels))
+			for _, label := range m.Labels {
+				row.Tags = append(row.Tags, Tag{
+					Key:   label.Name,
+					Value: label.Value,
+				})
+			}
+			switch mf.Type {
+			case COUNTER:
+				row.Value = m.Counter.Value
+				dst = append(dst, row)
+				tagsPool = append(tagsPool, row.Tags...)
+			case UNTYPED:
+				row.Value = m.Untyped.Value
+				dst = append(dst, row)
+				tagsPool = append(tagsPool, row.Tags...)
+			case SUMMARY:
+				dst = append(dst,
+					Row{
+						Metric: fmt.Sprintf("%s_count", mf.Name),
+						Value:  float64(m.Summary.SampleCount),
+					},
+					Row{
+						Metric: fmt.Sprintf("%s_sum", mf.Name),
+						Value:  m.Summary.SampleSum,
+					},
+				)
+				tagsPool = slices.Concat(tagsPool, row.Tags, row.Tags)
+				row.Tags = append(row.Tags, Tag{
+					Key: "quantile",
+				})
+				for _, q := range m.Summary.Quantiles {
+					row.Tags[tagsCount].Value = strconv.FormatFloat(q.Quantile, 'g', 3, 64)
+					row.Value = q.Value
+					dst = append(dst, row)
+					tagsPool = append(tagsPool, row.Tags...)
+				}
+			default:
```

Comment:
is it valid protobuf encoded message if we reached default case?

Reply:
in a future it'll support more formats (like histograms). don't know if it makes sense to generate an error here